web/docs/guides/websockets.md

```diff
@@ -0,0 +1,129 @@
+
+## Enable in Wasp file
+Here, we specify we are using WebSockets by adding `webSocket` to our `app` and providing the required `fn`. You can optionally specify a list of `entities` for use in your context (like in operations).
+
+```wasp title=todoApp.wasp
+app todoApp {
+  // ...
+
+  webSocket: {
+    fn: import { webSocketFn } from "@server/webSocket.js"
+    // Optional: entities: [...]
+  },,
+}
+```
+
+## Optional: Add shared types for client and server use
+If TypeScript is your thing, you can add shared interface definitions to use on both the client and server. This allows you to type all the messages you send and receive!
+
+```ts title=src/shared/webSocket.ts
+import { WaspSocketData } from '@wasp/universal/types'
+
+export interface ServerToClientEvents {
+  chatMessage: (msg: { id: string, username: string, text: string }) => void;
+}
+
+export interface ClientToServerEvents {
+  chatMessage: (msg: string) => void;
+}
+
+export interface InterServerEvents { }
+
+// Data that is attached to the socket.
+// NOTE: Wasp automatically injects the JWT into the connection,
+// and if present/valid, the server adds a user to the socket.
+export interface SocketData extends WaspSocketData { }
+```
+
+## Add server support
+On the server, you will get `io` argument and `context` for your WebSocket function. You can use this `io` object to register callbacks for all the regular Socket.IO events.
+
+```ts title=src/server/webSocket.ts
+import { v4 as uuidv4 } from 'uuid'
+import { WebSocketDefinition } from '@wasp/webSocket'
+import { ServerToClientEvents, ClientToServerEvents, InterServerEvents, SocketData } from '../shared/webSocket'
+
+export const webSocketFn: WebSocketDefinition<
+  ClientToServerEvents,
+  ServerToClientEvents,
+  InterServerEvents,
+  SocketData
+> = (io, context) => {
+  io.on('connection', (socket) => {
+    const username = socket.data.user?.email || socket.data.user?.username || 'unknown'
+    console.log('a user connected: ', username)
+
+    socket.on('chatMessage', async (msg) => {
+      console.log('message: ', msg)
+      io.emit('chatMessage', { id: uuidv4(), username, text: msg })
+    })
+  })
+}
+```
+
+## Add client support
+Client access to WebSockets is provided by the `useSocket` abstraction. It returns:
+- `socket: Socket` for sending and receiving events.
+- `isConnected: boolean` for showing a display of the Socket.IO connection status.
+  - Note: Wasp automatically connects establishes a WebSocket connection from the client to the server, so you do not need to explicitly `socket.connect()` or `socket.disconnect()`. All components using `useSocket` share the same underlying `socket`.
+- `refreshAuthToken: () => void` for refreshing the auth token on the `socket` when login status changes.
```

Comment:
Should we use this somewhere in our Auth logic?

Reply:
yes, on login for example.

Although, if we do the listener thing in api.ts, then that should be enough.